src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -106,8 +111,36 @@ public void createDataset(Dataset dataset) {
 
             bigQueryProject.createDataset(datasetName, dataset.getDescription());
             for (DatasetTable table : dataset.getTables()) {
-                bigQueryProject.createTable(datasetName, table.getRawTableName(), buildSchema(table, true));
-                bigQueryProject.createTable(datasetName, table.getSoftDeleteTableName(), buildSoftDeletesSchema());
+                PartitionStrategy partitionStrategy = table.getPartitionStrategy();
+                Column partitionColumn = table.getPartitionColumn();
+                Schema rawSchema = buildSchema(table, true);
+
+                if (partitionStrategy == null) {
+                    bigQueryProject.createUnpartitionedTable(datasetName, table.getRawTableName(), rawSchema);
+                } else {
+                    if (partitionStrategy == PartitionStrategy.INT_COLUMN) {
+                        IntPartitionOptionsModel options = table.getIntPartitionSettings();
+                        RangePartitioning.Range range = RangePartitioning.Range.newBuilder()
+                            .setStart(options.getMin())
+                            .setEnd(options.getMax())
+                            .setInterval(options.getInterval())
+                            .build();
+                        RangePartitioning partitioning = RangePartitioning.newBuilder()
+                            .setField(partitionColumn.getName())
+                            .setRange(range)
+                            .build();
+                        bigQueryProject.createRangePartitionedTable(
+                            datasetName, table.getRawTableName(), rawSchema, partitioning);
+                    } else {
+                        TimePartitioning partitioning = TimePartitioning.newBuilder(TimePartitioning.Type.DAY)
```

Comment:
What other types are there? Worth considering exposing this as an option to stewards now or in the future?

Reply:
There are no other types (yet?). I'll add a comment.